tasks/main.yml

```diff
@@ -1,23 +1,26 @@
 ---
-- name: Configure the Docker APT key
-  apt_key: keyserver="{{ docker_keyserver }}"
-           id="58118E89F3A912897C070ADBF76221572C52609D"
-           state=present
+- name: Check python version
+  shell: pip show python | grep Version | cut -d " " -f2
+  register: python_version
+
+- name: Install apt_key dependencies
+  pip: name={{ item }} state=present
+  with_items:
+    - urllib3
+    - pyOpenSSL
+    - ndg-httpsclient
+    - pyasn1
+  when: python_version.stdout | version_compare('2.7.9', '<')
+
+- name: Download Docker APT key
+  apt_key: url={{ docker_repository_url }}/gpg id=0EBFCD88 state=present
 
 - name: Configure the Docker APT repository
-  apt_repository: repo="deb https://apt.dockerproject.org/repo {{ ansible_distribution|lower }}-{{ ansible_distribution_release }} main"
+  apt_repository: repo="deb {{ docker_repository_url }} {{ ansible_distribution_release }} stable"
```

Comment:
According to the Docker docs, stable should always be installed, even if we want to install edge. I added a `docker_install_edge` variable to determine if we add the `edge` channel.